packages/core/src/execution-engine/partial-execution-utils/find-start-nodes.ts

```diff
@@ -111,7 +111,8 @@ function findStartNodesRecursive(
 		// If the node has multiple outputs, only follow the outputs that have run data.
 		const hasNoRunData =
 			nodeRunData === null || nodeRunData === undefined || nodeRunData.data.length === 0;
-		if (hasNoRunData) {
+		const hasNoPinnedData = !(pinData[outGoingConnection.from.name] !== undefined);
```

Comment:
Can we not combine the two negations?

```suggestion
		const hasNoPinnedData = pinData[outGoingConnection.from.name] === undefined;
```

or 


```suggestion
		const hasNoPinnedData = outGoingConnection.from.name in pinData;
```